pilot/pkg/dns/dns.go

```diff
@@ -328,6 +328,7 @@ func (table *LookupTable) lookupHost(qtype uint16, hostname string) ([]dns.RR, b
 // the lookup table, where N is number of search namespaces.
 func (table *LookupTable) buildDNSAnswers(altHosts map[string]struct{}, ipv4 []net.IP, ipv6 []net.IP, searchNamespaces []string) {
 	for h := range altHosts {
+		h = strings.ToLower(h)
```

Comment:
should we do the transformation higher up when we construct alt hosts?

Reply:
![image](https://user-images.githubusercontent.com/10192658/104030416-51c3cd00-5206-11eb-9224-d5cd100fd705.png)
If we do the transformation  generate alt hosts , we have to do it in every place where the host is generated. 
What we really need is to ensure that the  key of the table.allHosts is all lowercase, so i think is's better to do it here.

Reply:
Sorry i was thinking this was on the request path, not computed once. LGTM! thanks